### DISCUSSION TITLE
Fallback appender for Logback

### DISCUSSION THREAD

**Discussion by gminog** [qos-ch/logback#621] [Q&A]
Hello!
Copying from https://stackoverflow.com/questions/74877551/fallback-appender-for-logback:

> Let's assume that we are using an Appender that relies on a network connection. For instance, we may be using one of the following:
> - https://logback.qos.ch/apidocs/ch/qos/logback/classic/net/server/ServerSocketAppender.html
> - https://javadoc.io/static/net.logstash.logback/logstash-logback-encoder/6.2/net/logstash/logback/appender/AbstractLogstashTcpSocketAppender.html
>
> Since socket-based Appenders rely on the network, they are less reliable than a Console or File Appender.
> 
> Consider the scenario whereby the Socket-based Appender gets N errors or timeouts.
> It would be awesome if Logback could automatically switch to a File Appender until the network outage is resolved.
> 
> Does the native XML-based configuration of Logback allow us to define such a fallback policy or we have to implement something custom (e.g. https://www.baeldung.com/custom-logback-appender) ?

**Comment by gminog** [comment]
Hello!
Any pointers would be truly appreciated!

**Comment by ceki** [comment]
Logback does not offer a fallback policy off the shelf.

**Reply by gminog** [reply]
Thank you @ceki for your response.

Would it make sense to investigate us implementing something custom (e.g. https://www.baeldung.com/custom-logback-appender) using the existing SDK ? 
Or would that not be feasible, hence we should focus more on a PR for a core change in the SDK ?

We are totally willing to contribute with a PR, if needed!
